Dockerfile.local

```diff
@@ -39,63 +41,82 @@ RUN git clone -b v${MUT_VERSION} --depth 1 https://github.com/mongodb/mut.git \
 
 # install snooty frontend and docs-tools
 RUN git clone -b v${SNOOTY_FRONTEND_VERSION} --depth 1 https://github.com/mongodb/snooty.git       \
     && cd snooty                                                                                   \
     && npm ci --legacy-peer-deps --omit=dev                                                        
 
+RUN curl https://raw.githubusercontent.com/mongodb/docs-worker-pool/meta/makefiles/shared.mk -o shared.mk
+
+
+RUN git clone -b @dop/redoc-cli@${REDOC_CLI_VERSION} --depth 1 https://github.com/mongodb-forks/redoc.git redoc \
+    # Install dependencies for Redoc CLI
+    && cd redoc/ \
+    && npm ci --prefix cli/ --omit=dev
+    
+FROM initial as persistence
+
 RUN mkdir -p modules/persistence && chmod 755 modules/persistence
 COPY modules/persistence/package*.json ./modules/persistence/
 RUN cd ./modules/persistence \
     && npm ci --legacy-peer-deps 
-
-RUN  mkdir -p modules/oas-page-builder && chmod 755 modules/oas-page-builder
-COPY modules/oas-page-builder/package*.json ./modules/oas-page-builder/
-RUN cd ./modules/oas-page-builder \
-    && npm ci --legacy-peer-deps 
-
-# Root project build
-COPY package*.json ./
-RUN npm ci --legacy-peer-deps
 # Build persistence module
 
-COPY --chown=docsworker modules/persistence/tsconfig*.json ./modules/persistence
-COPY --chown=docsworker modules/persistence/src ./modules/persistence/src/
-COPY --chown=docsworker modules/persistence/index.ts ./modules/persistence
+COPY --chown=docsworker-xlarge modules/persistence/tsconfig*.json ./modules/persistence
+COPY --chown=docsworker-xlarge modules/persistence/src ./modules/persistence/src/
+COPY --chown=docsworker-xlarge modules/persistence/index.ts ./modules/persistence
 
 RUN cd ./modules/persistence \
     && npm run build
 
+FROM initial as oas
+
+RUN  mkdir -p modules/oas-page-builder && chmod 755 modules/oas-page-builder
+COPY modules/oas-page-builder/package*.json ./modules/oas-page-builder/
+RUN cd ./modules/oas-page-builder \
+    && npm ci --legacy-peer-deps 
 # Build modules
 # OAS Page Builder
-COPY --chown=docsworker modules/oas-page-builder/tsconfig*.json ./modules/oas-page-builder
-COPY --chown=docsworker modules/oas-page-builder/src ./modules/oas-page-builder/src/
-COPY --chown=docsworker modules/oas-page-builder/index.ts ./modules/oas-page-builder
+COPY --chown=docsworker-xlarge modules/oas-page-builder/tsconfig*.json ./modules/oas-page-builder
+COPY --chown=docsworker-xlarge modules/oas-page-builder/src ./modules/oas-page-builder/src/
+COPY --chown=docsworker-xlarge modules/oas-page-builder/index.ts ./modules/oas-page-builder
 
 RUN cd ./modules/oas-page-builder \
     && npm run build
 
+COPY --chown=docsworker-xlarge ./modules/oas-page-builder/dist/ ./modules/oas-page-builder
+
+FROM oas as root
+
+COPY --from=persistence --chown=docsworker-xlarge ${WORK_DIRECTORY}/modules/persistence/ ./modules/persistence
+COPY --chown=docsworker-xlarge ./modules/persistence/dist/ ./modules/persistence/
```

Comment:
Oh, I'm actually receiving an error from this line when running locally: 

```
Dockerfile.local:100
--------------------
  98 |     
  99 |     COPY --from=persistence --chown=docsworker-xlarge ${WORK_DIRECTORY}/modules/persistence/ ./modules/persistence
 100 | >>> COPY --chown=docsworker-xlarge ./modules/persistence/dist/ ./modules/persistence/
 101 |     
 102 |     # Root project build
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref c9d5839d-a372-45d3-aab0-f686e004deb8::jjcvfvfz7bc2nrmjzqquqor02: "/modules/persistence/dist": not found

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/skhd7adzlmqayhfural4uxgq8
```

Is the expectation here that we build the OAS module locally ourselves first?

Reply:
Hmm, but we build it on line 78...
@branberry Do we need line 99 anymore? Isn't line 100 doing "the needful" in this situation? Maybe 99 is actually a barrier for 100?

Reply:
Yeah, I'm a bit confused. Are you running into this issue locally? @mmeigs

Reply:
Just started running it locally a couple minutes ago to check! Sadly, it's been a second, so this will be a long cold build first.

Reply:
@rayangler very odd. Mine is failing about four lines later... oof. This was working yesterday. Great catch.

Reply:
@branberry I just commented out line 99 and it seems to be proceeding well!